docs/using_the_cli.md

```diff
@@ -0,0 +1,73 @@
+# Using the CLI
+
+The Style Dictionary command line interface (CLI) provides an executable system to create and act upon style dictionaries.
+
+
+# Installation
+To use the CLI, you can install it globally via npm:
+```bash
+$ npm install -g style-dictionary
+```
+
+
+# CLI Quick Start
+This will create a new folder called 'quick-start' and populate it with a Style Dictionary configuration and properties.
+```bash
+$ mkdir quick-start && cd quick-start
+$ style-dictionary init basic
+```
+You can then modify the properties in the properties directory and run the build command below. See the changes in the output generated in the build folder.
+```bash
+$ style-dictionary build
+```
+
+
+# Commands
+The CLI provides three basic commands:
+* (build [options])[#build]  Builds a style dictionary package from the current directory.
+* (clean [options])[#clean]  Removes files specified in the config of the style dictionary package of the current directory.
+* (init <type>)[#init]      Generates a starter style dictionary
+
+These commands can be run using:
+```bash
+$ style-dictionary [command] [options]
+```
+
+
+# build
+Builds a style dictionary package from the current directory. Usage:
+```bash
+$ style-dictionary build [options]
+```
+Options:
+| Name | Usage | Description |
+| :--- | :--- | :--- |
+| Configuration Path | -c <path>, --config <path> | Set the path to the configuration file. Defaults to './config.json'. |
+| Platform | -p <platform>, --platform <platform> | Only build a specific platform. If not supplied, builds all platform found in the configuration file. |
+
+
+# clean
+Removes files and folders generated by a previously run 'build' command. Usage:
+```bash
+$ style-dictionary clean [options]
+```
+Options:
+| Name | Usage | Description |
+| :--- | :--- | :--- |
+| Configuration Path | -c <path>, --config <path> | Set the path to the configuration file. Defaults to './config.json'. |
+| Platform | -p <platform>, --platform <platform> | Only clean a specific platform. If not supplied, cleans all platform found in the configuration file. |
+
+
+# init
+Generates a starter style dictionary, based on the supplied example type. Usage:
+```bash
+$ style-dictionary init <example-type>
+```
+Available examples can be found [here](https://github.com/amzn/style-dictionary/tree/master/examples/).
```

Comment:
Not true. You can install only a limited list of examples. See here: https://github.com/amzn/style-dictionary/blob/eba4970065941aab4403b4f544a805ae3846fcd9/bin/style-dictionary#L41

This will be even more true with the release of the new examples:
https://github.com/didoo/style-dictionary/blob/cba2672bcc31788b1d07b7d79ce7608b92e6b1b8/bin/style-dictionary#L41

Reply:
Urg...

Do we really want to not allow people to use a different starting example? I can change this, but I think it might make more sense to remove the code you are pointing to and have it do a check to see if the directory exists?

The types list could just be the directory list in the examples folder.

@dbanksdesign What do you think?

Reply:
The reasoning about this is that, if I am a user that needs to use the `init` command, probably it's the first time they I Style Dictionary. Giving me too many options can be counter-productive. As a first time user, probably what I want is to see/have some very basic setup, hence the "basic", or maybe I feel bolder and I try the "complete" to see how it looks like. If I have already started to use a little SD, probably I have read some documentation, maybe I had a look in the repo (and saw there is an "example" folder) and so on. So making the "init" a simple choice is better for me. 

Think about the "npm" example. If I really need to distribute my tokens as npm module (which means, I am using SD in a quite advanced way) the example will never be what I need. Is a good starting point, but the final result will be quite different. So, if you think to the two use cases (newbie and experienced) the "init" of an npm example can result confusing in the first case, and insufficient in the second case.

No strong opinions on this, just a gut feeling that in this case "less is better" :)

Reply:
For example, I have just discovered (and updated the example) that the "prepublish" script is deprecated, now is better to use "prepublishOnly". Our example was not "perfect" to be used as a real-world base for an npm module distribution.

Reply:
I see what you are saying. I guess my feelings are that the same argument could be applied to having examples at all - maybe we should only have the basic ones then. Or we should make sure we keep them up-to-date.

Maybe it is the command "init" causing some of the problem here. Really we mean "example".

Reply:
I think we should think about 'starter' projects as different than 'examples'. Maybe in the documentation we can say you can get started with the types of init projects with the CLI command, or take a look at all the examples. I think because we are breaking out our examples into single concepts like base64ing assets, npm module, etc. those are better served as reference/example code rather than a starter project. You would want to add that functionality after you get your Style Dictionary project up and running. Long-term, I think a scaffolding CLI should replace `init` anyways to give better starter projects than just copying a simple directory. Thoughts?

Reply:
> Long-term, I think a scaffolding CLI should replace init anyways to give better starter projects than just copying a simple directory. Thoughts?

This was my exact thinking and means that we should change this to a different command besides `init`. We use `example` for this and then build a simple `init` command as per #187.

Also, I'm confused about us thinking that we should only let people generate and use the `basic` examples. Examples exist to play with things and see how they work. Maybe I've used Style dictionary for 2 weeks and I understand the basics but now want to learn how base64ing assets work without risking messing up my current project - being able to generate and build an example gives me the playground I need to learn how to use it.

Reply:
Have an example command like: `style-dictionary example assets-base64-embed` and then turn `style-dictionary init` into the scaffolding script? I like that. Maybe let's wait until we get the scaffolding to make the change in documentation?

Reply:
Sounds good. I set up the documentation to reflect the current state.